File: examples/todo-java-spring/output/src/main/java/com/example/todo/FavouriteTodoService.java
package com.example.todo;

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
@Service
public class FavouriteTodoService {
    private final FavouriteTodoRepository repository;
    public FavouriteTodoService(FavouriteTodoRepository repository) {
        this.repository = repository;
    }
    public java.util.List<FavouriteTodo> findAll() {
        return repository.findAll();
    }
    public FavouriteTodo findById(java.util.UUID id) {
        java.util.Optional<FavouriteTodo> opt = repository.findById(id);
        if (!opt.isPresent()) {
            throw new java.util.NoSuchElementException("FavouriteTodo not found: " + id);
        }
        return opt.get();
    }
    @Transactional
    public FavouriteTodo create(FavouriteTodo entity) {
        entity.setCreatedAt(java.time.LocalDateTime.now());
        return repository.save(entity);
    }
    @Transactional
    public FavouriteTodo update(java.util.UUID id, FavouriteTodo entity) {
        FavouriteTodo existing = findById(id);
        existing.setName(entity.getName());
        existing.setDescription(entity.getDescription());
        existing.setUpdatedAt(java.time.LocalDateTime.now());
        return repository.save(existing);
    }
    @Transactional
    public void deleteById(java.util.UUID id) {
        repository.deleteById(id);
    }
    public java.util.List<FavouriteTodo> findFavourites() {
        return repository.findByFavouriteTrue();
    }
}

File: examples/todo-kotlin-quarkus/output/src/main/kotlin/com/example/todo/FavouriteTodoService.kt
package com.example.todo

import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject
import jakarta.transaction.Transactional
@ApplicationScoped
open class FavouriteTodoService {
    @Inject
    lateinit var repository: FavouriteTodoRepository
    fun findAll(): java.util.List<FavouriteTodo> {
        return repository.listAll()
    }
    fun findById(id: Long): FavouriteTodo? {
        return repository.findById(id)
    }
    @Transactional
    fun create(entity: FavouriteTodo): FavouriteTodo {
        entity.createdAt = java.time.LocalDateTime.now()
        repository.persist(entity)
        return entity
    }
    @Transactional
    fun update(id: Long, entity: FavouriteTodo): FavouriteTodo? {
        val existing = repository.findById(id) ?: return null
        existing.name = entity.name
        existing.description = entity.description
        existing.updatedAt = java.time.LocalDateTime.now()
        repository.persist(existing)
        return existing
    }
    @Transactional
    fun deleteById(id: Long): Boolean {
        return repository.deleteById(id)
    }
    fun findFavourites(): java.util.List<FavouriteTodo> {
        return repository.findFavourites()
    }
}